CMakeLists.txt

```diff
@@ -53,17 +57,58 @@ link_directories(${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
 
 find_package(Threads REQUIRED)
 
-# WolkAbout c++ SDK
+# WolkAbout C++ SDK, in their separated packages
+include_directories(core)
+file(GLOB INSTALL_HEADER_FILES)
+set(INSTALL_TARGETS)
 
-file(GLOB_RECURSE HEADER_FILES "${CMAKE_CURRENT_LIST_DIR}/core/*.h" "${CMAKE_CURRENT_LIST_DIR}/core/*.hpp")
-file(GLOB_RECURSE SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/core/*.cpp")
-add_library(${PROJECT_NAME} SHARED ${SOURCE_FILES})
+# Utilities
+file(GLOB UTILITY_HEADER_FILES core/utilities/*.h core/utilities/*.hpp core/Version.h)
+file(GLOB UTILITY_SOURCE_FILES core/utilities/*.cpp)
+
+add_library(${PROJECT_NAME}Utility SHARED ${UTILITY_SOURCE_FILES} ${UTILITY_HEADER_FILES})
+set_target_properties(${PROJECT_NAME}Utility PROPERTIES INSTALL_RPATH "$ORIGIN")
 
-target_link_libraries(${PROJECT_NAME} crypto paho-mqttpp3 paho-mqtt3as PocoCrypto PocoFoundation PocoNet PocoNetSSL Threads::Threads)
-set_target_properties(${PROJECT_NAME} PROPERTIES INSTALL_RPATH "$ORIGIN")
+list(APPEND INSTALL_HEADER_FILES ${UTILITY_HEADER_FILES})
+list(APPEND INSTALL_TARGETS ${PROJECT_NAME}Utility)
 
-add_dependencies(${PROJECT_NAME} pahomqttcpp libpoco)
+# Build the MQTT logic if Paho is enabled
+if (BUILD_PAHO_MQTT)
+    file(GLOB CORE_HEADER_FILES core/*.h)
+    file(GLOB CONNECTIVITY_HEADER_FILES core/connectivity/*.h)
+    file(GLOB MODEL_HEADER_FILES core/model/*.h)
+    file(GLOB PERSISTENCE_HEADER_FILES core/persistence/*.h)
+    file(GLOB PROTOCOL_HEADER_FILES core/protocol/*.h)
+    file(GLOB SERVICE_HEADER_FILES core/service/*.h)
+    file(GLOB MQTT_SOURCE_FILES core/connectivity/*.cpp
+            core/model/*.cpp
+            core/persistence/*.cpp
+            core/protocol/*.cpp
+            core/service/*.cpp)
 
-target_include_directories(${PROJECT_NAME} PUBLIC "${CMAKE_CURRENT_LIST_DIR}/core")
+    add_library(${PROJECT_NAME} SHARED ${MQTT_SOURCE_FILES} ${CORE_HEADER_FILES} ${CONNECTIVITY_HEADER_FILES}
+            ${MODEL_HEADER_FILES} ${PERSISTENCE_HEADER_FILES} ${PROTOCOL_HEADER_FILES} ${SERVICE_HEADER_FILES})
+    target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}Utility paho-mqttpp3 paho-mqtt3as)
+    set_target_properties(${PROJECT_NAME} PROPERTIES INSTALL_RPATH "$ORIGIN")
+    add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}Utility pahomqttc pahomqttcpp)
 
+    list(APPEND INSTALL_HEADER_FILES ${MQTT_HEADER_FILES})
+    list(APPEND INSTALL_TARGETS ${PROJECT_NAME})
+endif ()
```

Comment:
endif what? 

endif(BUILD_PAHO_MQTT)

Reply:
The endif() condition is not necessary, it is only in CMake for backwards compatibility reasons. [Link to documentation](https://cmake.org/cmake/help/latest/command/endif.html).

Even next to that, all the if statements are only followed by endif, and not if, they are all 1-depth ifs, no-body can get lost in this.

Reply:
If so, then remove this endif (NOT DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY) with endif()

Reply:
Done.